pkg/controller/volume/scheduling/scheduler_binder.go

```diff
@@ -248,20 +254,10 @@ func (b *volumeBinder) FindPodVolumes(pod *v1.Pod, node *v1.Node) (reasons Confl
 		b.podBindingCache.UpdateBindings(pod, node.Name, matchedBindings, provisionedClaims)
 	}()
 
-	// The pod's volumes need to be processed in one call to avoid the race condition where
-	// volumes can get bound/provisioned in between calls.
-	boundClaims, claimsToBind, unboundClaimsImmediate, err := b.getPodVolumes(pod)
-	if err != nil {
-		return nil, err
-	}
-
-	// Immediate claims should be bound
-	if len(unboundClaimsImmediate) > 0 {
-		return nil, fmt.Errorf("pod has unbound immediate PersistentVolumeClaims")
-	}
-
 	// Check PV node affinity on bound volumes
 	if len(boundClaims) > 0 {
+		// TODO if node affinity does not match, we should
+		// UnschedulableAndUnresolvable error back to scheduler framework
```

Comment:
UnschedulableAndUnresolvable error is more meaningful than internal error, this can be in a separate PR

Reply:
made a mistake here, it's already an UnschedulableAndUnresolvable error now